### DISCUSSION TITLE
Adding a node within the viewable area where no other node is located

### DISCUSSION THREAD

**Discussion by CodyBontecou** [bcakmakoglu/vue-flow#1390] [Q&A]
I'm curious if you can help me with this particular problem.

Currently, adding new nodes can overlap one another. I'd like to add a node in the viewable area, but ideally where no other node is, or at least in a way the user can see the node without it being overlapped or hidden.

**Comment by bcakmakoglu** [comment]
What exactly are you having trouble with though? 😅

You could loop over your nodes and check what position is still available and add a node there, all the info you would need is available on nodes (position + dimensions).
